tests/vmi_multus_test.go

```diff
@@ -662,27 +658,16 @@ var _ = Describe("Multus", func() {
 				Expect(err).ToNot(HaveOccurred(), "Should create VMI successfully")
 				tests.WaitForSuccessfulVMIStart(agentVMI)
 
-				getOptions := &metav1.GetOptions{}
-				var updatedVmi *v1.VirtualMachineInstance
-
-				// Need to wait for cloud init to finnish and start the agent inside the vmi.
-				Eventually(func() bool {
-					updatedVmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(agentVMI.Name, getOptions)
-					Expect(err).ToNot(HaveOccurred())
-					for _, condition := range updatedVmi.Status.Conditions {
-						if condition.Type == "AgentConnected" && condition.Status == "True" {
-							return true
-						}
-					}
-					return false
-				}, 420*time.Second, 2).Should(BeTrue(), "Should have agent connected condition")
+				// Need to wait for cloud init to finish and start the agent inside the vmi.
+				tests.WaitAgentConnected(virtClient, agentVMI)
 
+				getOptions := &metav1.GetOptions{}
 				Eventually(func() bool {
-					updatedVmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(agentVMI.Name, getOptions)
+					updatedVmi, _ := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(agentVMI.Name, getOptions)
 					return len(updatedVmi.Status.Interfaces) == 4
```

Comment:
I know this is not a regression, but it looks like this would cause null pointer exception in case VirtualMachineInstance failed. Maybe we can just `continue` in case err != nil?

Reply:
+1

Reply:
This has been addressed on line 667, right?

Reply:
I don't think. in the 667 we just get the vmi object but we need to check if the interfaces array inside status is not nil.

Reply:
I'm not sure how can this happen as at this stage the vm should be running (we've passed tests.WaitForSuccessfulVMIStart(agentVMI)) so there should be at least one interface... but I guess it doesn't hurt checking... @booxter what do you think?

Reply:
I should have pushed against fixing the original issue with not checking the error at all; if I would do it, I wouldn't get more requests to fix unrelated code. :)

Reply:
@vladikr @SchSeba I have no plan to update the patch unless someone explicitly says they will block it without me fixing this unrelated issue. Does anyone of you plan to block this PR for the issue being discussed?